src/utils/vm_mount.cpp

```diff
@@ -105,12 +107,44 @@ mp::VMMount::VMMount(const std::string& sourcePath,
     if (errors.size())
         throw std::runtime_error(
             fmt::format("Mount cannot apply mapping with duplicate ids:{}", fmt::to_string(errors)));
+
+    resolve_source_path();
 }
 
 mp::VMMount::VMMount(const QJsonObject& json) : VMMount{parse_json(json)} // delegate on copy ctor
 {
 }
 
+void mp::VMMount::resolve_source_path()
+{
+    std::error_code err;
+    fs::path path{source_path};
+
+    auto status = MP_FILEOPS.symlink_status(path, err);
+
+    if (status.type() == fs::file_type::symlink)
+    {
+        auto symlink_path = MP_FILEOPS.read_symlink(path, err);
+
+        if (err)
+            throw std::runtime_error(
+                fmt::format("Mount symlink source path \"{}\" could not be read: {}.", source_path, err.message()));
+
+        if (symlink_path.is_relative())
+            symlink_path = path.parent_path() / symlink_path;
+
+        path = fs::weakly_canonical(symlink_path, err);
+
+        if (err)
+            throw std::runtime_error(
+                fmt::format("Mount symlink source path \"{}\" could not be made weakly canonical: {}.",
```

Comment:
This would end up being shown to the user, right? If so, I would try to make it more user friendly. Perhaps:

```suggestion
                fmt::format("Could not resolve symlinks in mount source path \"{}\": {}.",
```